spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/core/AppDeploymentRequest.java

```diff
@@ -92,12 +94,8 @@ public AppDeploymentRequest(AppDefinition definition, Resource resource,
 		Assert.notNull(resource, "resource must not be null");
 		this.definition = definition;
 		this.resource = resource;
-		this.deploymentProperties = deploymentProperties == null
-				? Collections.<String, String>emptyMap()
-				: Collections.unmodifiableMap(deploymentProperties);
-		this.commandlineArguments = commandlineArguments == null
-				? Collections.<String>emptyList()
-				: Collections.unmodifiableList(commandlineArguments);
+		this.deploymentProperties = deploymentProperties;
```

Comment:
But, what happens if this constructor `public AppDeploymentRequest(AppDefinition definition, Resource resource,Map<String, String> deploymentProperties, List<String> commandlineArguments)` is 
`directly` invoked with `null` deploymentProperties and commandlineArguments ?